airbyte-db/db-lib/src/main/java/io/airbyte/db/factory/DataSourceFactory.java

```diff
@@ -196,19 +198,23 @@ private DataSourceBuilder() {}
      *
      * @param connectionProperties custom jdbc_url_parameters containing information on connection
      *        properties
+     * @param driverClassName name of the JDBC driver
      * @return DataSourceBuilder class used to create dynamic fields for DataSource
      */
-    private static long getConnectionTimeoutMs(final Map<String, String> connectionProperties) {
-      final Duration connectionTimeout;
-      // TODO: the usage of CONNECT_TIMEOUT_KEY is Postgres specific, may need to extend for other
-      // databases
-      connectionTimeout =
-          connectionProperties.containsKey(CONNECT_TIMEOUT_KEY) ? Duration.ofSeconds(Long.parseLong(connectionProperties.get(CONNECT_TIMEOUT_KEY)))
-              : CONNECT_TIMEOUT_DEFAULT;
-      if (connectionTimeout.getSeconds() == 0) {
-        return connectionTimeout.toMillis();
+    private static long getConnectionTimeoutMs(final Map<String, String> connectionProperties, String driverClassName) {
+      // TODO: the usage of CONNECT_TIMEOUT is Postgres specific, may need to extend for other databases
+
+      if (driverClassName.equals(DatabaseDriver.POSTGRESQL.getDriverClassName())) {
+        final String pgPropertyConnectTimeout = CONNECT_TIMEOUT.getName();
+        // If the PGProperty.CONNECT_TIMEOUT was set by the user, then take its value, if not take the
+        // default
+        return (connectionProperties.containsKey(pgPropertyConnectTimeout)
+            && (Long.parseLong(connectionProperties.get(pgPropertyConnectTimeout)) > 0))
+                ? Duration.ofSeconds(Long.parseLong(connectionProperties.get(pgPropertyConnectTimeout))).toMillis()
+                : Duration.ofSeconds(Long.parseLong(Objects.requireNonNull(CONNECT_TIMEOUT.getDefaultValue()))).toMillis();
+
       } else {
-        return (connectionTimeout.compareTo(CONNECT_TIMEOUT_DEFAULT) > 0 ? connectionTimeout : CONNECT_TIMEOUT_DEFAULT).toMillis();
+        return CONNECT_TIMEOUT_DEFAULT.toMillis();
```

Comment:
If I'm understanding this correctly this changes the default behavior for all other connectors where the user can pass in a `connection_timeout` property. In the javadoc description, we allow for users to set the `connection_timeout` to 0 which means timeout duration is infinity

This seems to change that behavior for all connectors that isn't postgres. Also as a note the reason why default of 60 seconds was chosen in the past was due to integration tests failing on the default of 30 seconds (also in the javadoc description). If that has changed, then updating the javadoc comment to match makes sense

Reply:
> If I'm understanding this correctly this changes the default behavior for all other connectors where the user can pass in a `connection_timeout` property. In the javadoc description, we allow for users to set the `connection_timeout` to 0 which means timeout duration is infinity
> 
> This seems to change that behavior for all connectors that isn't postgres. Also as a note the reason why default of 60 seconds was chosen in the past was due to integration tests failing on the default of 30 seconds (also in the javadoc description). If that has changed, then updating the javadoc comment to match makes sense

@ryankfu good point 
changed logic on 
`
if (driverClassName.equals(DatabaseDriver.POSTGRESQL.getDriverClassName())) {
       new behaviour ;
  } else {
       old behaviour ;
  }
`